### PR TITLE
feature/REDBOX-171-add-user-filtering-to-core-api-part-2

### PR DESCRIPTION
**Pull request by gecBurton**
## Context

As a User I want to be able to query my documents but no one elseses

## Changes proposed in this pull request

* I have added a filter to the retriever
* I have expanded the integration tests to cover 2 users

## Relevant links

https://technologyprogramme.atlassian.net/browse/REDBOX-171

## Things to check

- [ ] I have added any new ENV vars in all deployed environments
- [x] I have tested any code added or changed
- [x] I have run integration tests https://github.com/i-dot-ai/redbox-copilot/actions/runs/8999572233

### CODE REVIEW

File: core_api/src/routes/chat.py [hunk 1]
```diff
@@ -144,7 +155,9 @@ def rag_chat(chat_request: ChatRequest, _user_uuid: Annotated[UUID, Depends(get_
 
     standalone_question = condense_question_chain({"question": question, "chat_history": previous_history})["text"]
 
-    docs = vector_store.as_retriever().get_relevant_documents(standalone_question)
+    docs = vector_store.as_retriever(
+        search_kwargs={"filter": {"term": {"creator_user_uuid.keyword": str(user_uuid)}}}
```
Comment (gecBurton) [inline core_api/src/routes/chat.py:159]: this is the key change

File: django_app/redbox_app/redbox_core/views.py [hunk 2]
```diff
@@ -156,10 +156,9 @@ def remove_doc_view(request, doc_id: str):
 
 @login_required
 def sessions_view(request, session_id: str = ""):
-    
     USE_STREAMING = False
     STREAMING_ENDPOINT = "ws://localhost:8888"
-    
+
```
Comment (gecBurton) [inline django_app/redbox_app/redbox_core/views.py:161]: ruff

File: tests/test_e2e.py [hunk 3]
```diff
@@ -1,27 +1,34 @@
...
+from uuid import uuid4, UUID
 
 import pytest
 import requests
+from jose import jwt
+
 
 # TODO: add e2e tests involving the Django app, checking S3 upload
 
+USER_UUIDS: list[UUID] = [uuid4(), uuid4()]
```
Comment (rachaelcodes) [inline tests/test_e2e.py:12]: Nice parameterization with these :)

**Review by rachaelcodes — APPROVED** [review]
LGTM - helpful comments

**Pull request merged**